transport/http2_client.go

```diff
@@ -413,7 +414,7 @@ func (t *http2Client) NewStream(ctx context.Context, callHdr *CallHdr) (_ *Strea
 	t.hEnc.WriteField(hpack.HeaderField{Name: ":scheme", Value: t.scheme})
 	t.hEnc.WriteField(hpack.HeaderField{Name: ":path", Value: callHdr.Method})
 	t.hEnc.WriteField(hpack.HeaderField{Name: ":authority", Value: callHdr.Host})
-	t.hEnc.WriteField(hpack.HeaderField{Name: "content-type", Value: "application/grpc"})
+	t.hEnc.WriteField(hpack.HeaderField{Name: "content-type", Value: s.contentType})
```

Comment:
I think this should stay as `"application/grpc"` by default, as this is the existing behavior. If we change this, this could have unintended consequences for existing users, even if this means some grpc implementations are not handling the `Content-Type` header properly.